insights/parsers/package_provides_java.py

```diff
@@ -0,0 +1,36 @@
+"""
+PackageProvidesJava - Command ``rpm -qf``
+=========================================
+
+This parser handles ``rpm -qf`` output.
+
+Typical contents of the ``rpm -qf package`` command::
+
+    java-1.8.0-openjdk-1.8.0.144-0.b01.el7_4.x86_64
+
+Examples:
```

Comment:
Java has many different versions and is so flexible that some of them can be installed at the same time on the same system. Some customers might not use the default java (/usr/bin/java) for some special application, we have to get the full path of each specific java process and use `rpm -qf` to check which package it belongs to.

 And since `readlink` can only check one symbol line once, so we have to put it into a loop.

Currently, we'd like to use the following pre_command to get the 
~~~
for jp in `/bin/ps auxwww | grep java | grep '\-Dcatalina.base'| awk '{print $11}' | sort -u`; do readlink -e $jp; done | sort -u
~~~

I totally agree with you that we'd better add more detail documentation to explain this. 

@zhangyi733

Reply:
That was the source of my confusion, if the doc is clearer as to the purpose of this parser then that would help significantly.